src/components/PostDetails.tsx

```diff
@@ -58,35 +51,22 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
   */
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
+    await dispatch(
+      commentsActions.addComment({
         name,
         email,
         body,
         postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
+      }),
+    );
   };
 
   const deleteComment = async (commentId: number) => {
     // we delete the comment immediately so as
     // not to make the user wait long for the actual deletion
     // eslint-disable-next-line max-len
-    setComments(currentComments =>
-      currentComments.filter(comment => comment.id !== commentId),
-    );
-
-    await commentsApi.deleteComment(commentId);
+    dispatch(commentsActions.deleteCurrentComment(commentId));
+    await dispatch(commentsActions.removeComment(commentId));
```

Comment:
The `deleteComment` function dispatches two actions: `deleteCurrentComment` and `removeComment`. Ensure both are necessary for your application's logic, as this might lead to redundant operations.